packages/ui/src/ui/pages/navigation/NavigationDescription/NavigationDescription.tsx

```diff
@@ -51,28 +52,19 @@ const NavigationDescription: FC<Props> = ({className}) => {
         );
     }, [dispatch, expanded]);
 
-    const handleEditClick = useCallback(() => {
-        oldValue.current = annotation;
-        dispatch({type: SET_ANNOTATION_EDITING, data: true});
-    }, [annotation, dispatch]);
-
-    const handleCancelClick = useCallback(() => {
-        dispatch({type: SET_ANNOTATION, data: oldValue.current});
-        dispatch({type: SET_ANNOTATION_EDITING, data: false});
-    }, [dispatch]);
-
-    const handleChangeDescription = useCallback(
-        ({value}: {value: string | undefined}) => {
-            dispatch({type: SET_ANNOTATION, data: value || ''});
-        },
-        [dispatch],
-    );
-
-    const handleSaveClick = useCallback(() => {
-        dispatch(saveAnnotation(path));
-    }, [dispatch, path]);
+    const handleChangeDescription = ({value}: {value: string | undefined}) => {
+        dispatch(
+            ytApi.util.updateQueryData('fetchBatch', makeAnnotationFetchParams(path), () => [
+                {
+                    output: value,
+                },
+            ]),
```

Comment:
Does the code replace loaded data from api by user input?
It is better to keep user input separately from loaded data.